src/background.js

```diff
@@ -1,24 +1,72 @@
-let color = '#3aa757';
-
 chrome.runtime.onInstalled.addListener(() => {
-  chrome.storage.sync.set({ color });
-  console.log('Default background color set to %cbanana', `color: ${color}`);
-  setUp();
+  run();
 });
 
 chrome.runtime.onStartup.addListener(() => {
-  setUp();
+  run();
 });
 
 
-function setUp() {
-    console.log("Starting up crawler extension");
-//    chrome.alarms.onAlarm.addListener(runCrawlIteration);
-//    chrome.alarms.create("crawlAlarm", {delayInMinutes: 1.0/60, periodInMinutes: 0.1});
-    setInterval(runCrawlIteration, 1000);
+function run() {
+  crawler = new Crawler();
+  crawler.setUp();
 }
 
 
-function runCrawlIteration() {
-  console.log("Running crawl iteration");
+class Crawler {
+  constructor() {
+    this.curatedDomains = [];
+  }
+
+  loadCuratedDomains() {
+    const url = chrome.runtime.getURL('../../assets/data/hn-top-domains.json');
+    fetch(url).then(response => {
+      response.json().then(data => {
+        this.curatedDomains = Object.keys(data);
+        console.log("Loaded curated domains", this.curatedDomains);
+      });
+    });
+  }
+
+  chooseDomain() {
+    time = new Date().getTime();
```

Comment:
Do not forget to use `let` or `const` when you declare your variables. For example, here it is a `const`.